unumpy/_multimethods.py

```diff
@@ -1132,6 +1256,25 @@ def column_stack(tup):
     return tup
 
 
+def _dstack_default(tup):
+    arrays = []
+    for arr in tup:
+        if ndim(arr) == 1:
+            arrays.append(expand_dims(arr, (0, -1)))
+        elif ndim(arr) == 2:
+            arrays.append(expand_dims(arr, -1))
+        else:
+            arrays.append(arr)
+
+    return concatenate(arrays, axis=2)
+
+
+@create_numpy(_first_argreplacer, default=_dstack_default)
```

Comment:
`_self_argreplacer`?

Reply:
I still confuse the two often as I don't understand exactly the difference between them. From what I understand `_first_argreplacer` is to be used when we want to dispatch an argument that's a tuple of arrays but I'm probably wrong here. The other stack multimethods use `_first_argreplacer` as well.